sklearn/utils/estimator_checks.py

```diff
@@ -625,8 +715,9 @@ def check_estimators_partial_fit_n_features(name, Alg):
     assert_raises(ValueError, alg.partial_fit, X[:, :-1], y)
 
 
-def check_clustering(name, Alg):
-    X, y = make_blobs(n_samples=50, random_state=1)
+def check_clustering(name, Alg, readonly=False):
+    X, y = _make_blobs_with_mode(n_samples=50, random_state=1, readonly=readonly)
+    # TODO: this breaks read only mode
```

Comment:
Line behind reads ```python     X, y = shuffle(X, y, random_state=7)```, which triggers a copy. I could put it into `_make_blobs_with_mode` but would make it even more hacky